tools/clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -7665,6 +7665,9 @@ def err_hlsl_unsupported_builtin_op: Error<
 def warn_hlsl_builtin_constant_unavailable: Warning<
   "potential misuse of built-in constant %0 in shader model %1; introduced"
   " in shader model %2">, InGroup<HLSLAvailabilityConstant>;
+def warn_hlsl_builtin_type_unavailable: Warning<
+  "potential misuse of built-in type %0 in shader model %1; introduced"
+  " in shader model %2">, InGroup<HLSLAvailability>;
```

Comment:
Should we add `DefaultError` to make this an error by default?

Reply:
> Should we add DefaultError to make this an error by default?

I don't think so given that the type may be used on dead code paths in HLSL and optimized out before it is caught by the DXIL validator. The DXIL validator change we track in #7234

Reply:
Turned this into a `DefaultError` - we do the same for the other diags of this kinds (except bui;ltin constants).Resolved,